File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/05_Coins.js
function coins(input) {
    // 2 лева, 1 лев, 50 стотинки, 20 стотинки, 10 стотинки, 5 стотинки, 2 стотинки или 1 стотинка 
    let moneyAmount = Number(input[0]);
    let coinsCount = 0;

    let currentAmountInCents = Math.round(moneyAmount * 100);
    // let currentAmountInCents = moneyAmount * 100;
    // console.log(currentAmountInCents);

    while (currentAmountInCents > 0) {  // while (cond1) {}
        coinsCount++;

        if (currentAmountInCents >= 200) {
            currentAmountInCents -= 200;
        } else if (currentAmountInCents >= 100) {
            currentAmountInCents -= 100;
        } else if (currentAmountInCents >= 50) {
            currentAmountInCents -= 50;
        } else if (currentAmountInCents >= 20) {
            currentAmountInCents -= 20;
        } else if (currentAmountInCents >= 10) {
            currentAmountInCents -= 10;
        } else if (currentAmountInCents >= 5) {
            currentAmountInCents -= 5;
        } else if (currentAmountInCents >= 2) {
            currentAmountInCents -= 2;
        } else if (currentAmountInCents >= 1) {
            currentAmountInCents -= 1;
        }
    }
    console.log(coinsCount);
}


// coins(["1.23"]);
// coins(["2.49"]);
// coins(["1.51"]);

File: Python_Courses/Python_Basics/Python_PB_2024_01/05_While_Loop_Exercise/05_Coins.py
# [2lv, 1lv, 50st, 20st, 10st, 5st, 2st, 1st]

change = float(input())

coins = 0
change_left = change
while change_left > 0:
    if change_left >= 2:
        change_left -= 2
    elif change_left >= 1:
        change_left -= 1
    elif change_left >= 0.50:
        change_left -= 0.50
    elif change_left >= 0.20:
        change_left -= 0.20
    elif change_left >= 0.10:
        change_left -= 0.10
    elif change_left >= 0.05:
        change_left -= 0.05
    elif change_left >= 0.02:
        change_left -= 0.02
    elif change_left >= 0.01:
        change_left -= 0.01

    coins += 1
    change_left = round(change_left, 2)

print(coins)
